### PR TITLE
Samuel Cha - Processes

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: ex1/ex1.c [hunk 1]
```diff
@@ -8,7 +8,44 @@
...
 int main(int argc, char *argv[])
 {
-    // Your code here
+    int x = 100;
+
+    int rc = fork();
+
+    if (rc < 0)
+    {
+        fprintf(stderr, "fork() failed\n");
```
Comment (seanchen1991) [inline ex1/ex1.c]: Don't forget to also `exit(1)` the program on a fork failure.

File: ex1/ex1.c [hunk 2]
```diff
@@ -8,7 +8,44 @@
...
+    }
+    else
+    {
+        printf("Parent int x before= %d\n", x);
+        x = 10;
+        printf("Parent int x after= %d\n", x);
+    }
+
+    /*
+        1) The value of var x is the same in both child and parent.
```
Comment (seanchen1991) [inline ex1/ex1.c]: Awesome notes! 👍

File: ex2/ex2.c [hunk 3]
```diff
@@ -1,14 +1,63 @@
...
 
-int main(int argc, char* argv[])
+int main(int argc, char *argv[])
 {
-    // Your code here 
-    
+    char text[100];
+    FILE *fp;
+
+    fp = fopen("text.txt", "r");
```
Comment (seanchen1991) [inline ex2/ex2.c]: The intention of this exercise was to have you write to the file, but I think I forgot to remove the contents from the `text.txt` file, so this isn't that surprising that you'd interpret the instructions this way.

File: ex3/ex3.c [hunk 4]
```diff
@@ -6,9 +6,22 @@
...
 
-int main(int argc, char* argv[])
+int main(int argc, char *argv[])
 {
-    // Your code here
+    int rc = fork();
 
+    if (rc < 0)
+    {
+        fprintf(stderr, "fork() failed\n");
```
Comment (seanchen1991) [inline ex3/ex3.c]: You'll want to `exit(1)` here.

File: ex5/ex5.c [hunk 5]
```diff
@@ -1,21 +1,54 @@
...
+        write(p[1], msg2, MSGSIZE);
+        write(p[1], msg3, MSGSIZE);
+    }
+
+    else
+    {
+        close(p[1]); /* close write pipe */
+
+        char inbuf[MSGSIZE];
+        int nbytes;
```
Comment (seanchen1991) [inline ex5/ex5.c]: In retrospect, this `nbytes` variable really isn't being used. In the `while` loop, we could just write it like this
```
while (read(p[0], inbuf, MSGSIZE) > 0)
```
to check that bytes are being read from the `read` call.

**Review by seanchen1991 — COMMENTED** [review]
Nice work. I like the notes you kept ✅

File: ex5/ex5.c (hunk 5 above)
Comment (samscha) [reply ex5/ex5.c]: 👍

**Comment by samscha** [comment]
comments noted

File: ex2/ex2.c (hunk 3 above)
Comment (samscha) [reply ex2/ex2.c]: oh my.. whoops

**Pull request closed without merging**